finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/ServerBridge.scala

```diff
@@ -15,6 +17,17 @@ private[netty4] class ServerBridge[In, Out](
 
   def initChannel(ch: Channel): Unit = {
     val transport: Transport[In, Out] = transportFac(ch)
-    serveTransport(transport)
+    val sslHandler = ch.pipeline().get(classOf[SslHandler])
+    if(sslHandler != null ) {
+      sslHandler.handshakeFuture().addListener( new GenericFutureListener[NettyFuture[Channel]] {
+        override def operationComplete(future: NettyFuture[Channel]): Unit = {
+          if(future.isSuccess) {
```

Comment:
we should make sure we handle non-success cases (interrupt, failure) gracefully here.

also, is it possible this will hang?  if I remember correctly, the sslhandler has a default timeout of 10 seconds?  is that right?